netsync/manager_test.go

```diff
@@ -0,0 +1,776 @@
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Attempt to process transaction
+	syncChan := make(chan struct{})
+	syncMgr.QueueTx(tx1, localNode, syncChan)
+	select {
+	case <-syncChan:
+	case <-time.After(time.Second):
+		t.Fatal("Timeout waiting for sync manager to process transaction 1")
+	}
+
+	// Assert transaction was accepted by checking that call was made to notify
+	// peers of the new transaction.
+	select {
+	case call := <-ctx.peerNotifier.announceNewTransactionsChan:
+		if len(call.newTxs) != 1 ||
+			!call.newTxs[0].Tx.Hash().IsEqual(tx1.Hash()) {
+
+			t.Fatalf("PeerNotifier received unexpected "+
+				"AnnounceNewTransactions call: %v", call.newTxs)
+		}
+	default:
+		t.Fatalf("Expected SyncManager to make " +
+			"AnnounceNewTransactions call to PeerNotifier")
+	}
+
+	// Create chain of two transactions and broadcast out of order
+	tx2, err := createSpendingTx(tx1, 0, scriptSig, address)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tx3, err := createSpendingTx(tx2, 0, scriptSig, address)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Attempt to process orphan transaction
+	syncMgr.QueueTx(tx3, localNode, syncChan)
+	select {
+	case <-syncChan:
+	case <-time.After(time.Second):
+		t.Fatal("Timeout waiting for sync manager to process transaction 3")
+	}
+
+	// AnnounceNewTransactions should be called with no transactions
+	select {
+	case call := <-ctx.peerNotifier.announceNewTransactionsChan:
+		if len(call.newTxs) != 0 {
+			t.Fatalf("PeerNotifier received unexpected "+
+				"AnnounceNewTransactions call: %v", call.newTxs)
+		}
+	default:
+		t.Fatalf("Expected SyncManager to make " +
+			"AnnounceNewTransactions call to PeerNotifier")
+	}
+
+	// Now process parent transaction
+	syncMgr.QueueTx(tx2, localNode, syncChan)
+	select {
+	case <-syncChan:
+	case <-time.After(time.Second):
+		t.Fatal("Timeout waiting for sync manager to process transaction 2")
+	}
+
+	// AnnounceNewTransactions should be called with both orphan and parent
+	select {
+	case call := <-ctx.peerNotifier.announceNewTransactionsChan:
+		if len(call.newTxs) != 2 ||
+			!call.newTxs[0].Tx.Hash().IsEqual(tx2.Hash()) ||
+			!call.newTxs[1].Tx.Hash().IsEqual(tx3.Hash()) {
+
+			t.Fatalf("PeerNotifier received unexpected "+
+				"AnnounceNewTransactions call: %v", call.newTxs)
+		}
+	default:
+		t.Fatalf("Expected SyncManager to make " +
+			"AnnounceNewTransactions call to PeerNotifier")
+	}
+
+	// Send non-standard, invalid transaction with corrupt scriptSig
+	badScriptSig := append(scriptSig, 0xaa, 0xbb, 0xcc, 0xdd)
+	tx4, err := createSpendingTx(tx3, 0, badScriptSig, address)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Attempt to process orphan transaction
+	syncMgr.QueueTx(tx4, localNode, syncChan)
+	select {
+	case <-syncChan:
+	case <-time.After(time.Second):
+		t.Fatal("Timeout waiting for sync manager to process transaction 4")
+	}
+
+	// AnnounceNewTransactions should not be called for invalid transaction
```

Comment:
Given the discussion above, if that is changed as suggested, this would be `for an invalid non-orphan transaction` to indicate that it is indeed testing the desired property.